prow/def.bzl

```diff
@@ -16,13 +16,24 @@ load("@io_bazel_rules_docker//container:image.bzl", "container_image")
 load("@io_bazel_rules_docker//container:bundle.bzl", "container_bundle")
 load("@io_bazel_rules_docker//contrib:push-all.bzl", "container_push")
 load("@io_bazel_rules_docker//go:image.bzl", "go_image")
+load("@io_bazel_rules_go//go:def.bzl", "go_binary")
 load("@io_bazel_rules_k8s//k8s:object.bzl", "k8s_object")
 load("@io_bazel_rules_k8s//k8s:objects.bzl", "k8s_objects")
 load(
     "//def:image.bzl",
     _image_tags = "tags",
 )
 
+## prow_binary is a macro for creating a binary target with injected vars
+def prow_binary(
+        name,
+        **kwargs):
+    go_binary(
+        name = name,
+        x_defs = {"k8s.io/test-infra/prow/version.Name": name},
```

Comment:
I believe you want this in the `prow_image()` (which embeds `:go_default_library`, not the binary target).

In which case I don't really think we need a `prow_binary()` rule?

https://github.com/bazelbuild/rules_docker/blob/6b477e319e09b2c4a8ce1ecfd5df7bf3a15e88fa/go/image.bzl#L108

Reply:
Why wouldn't we want the binaries to know who they are without having to be embedded in an image?

Reply:
We want them to know who they are when they are embedded in an image. Right now they don't.

Reply:
Ooooooh ... I see. Bazel!